update some triage steps, remove `requires-incomplete-features` (#986)
rust-lang-nursery/rust-forge: src/release/issue-triaging.md

- Add `A-*` labels based on the code that causes the issue (check backtraces!), and the nature of the repro (e.g. if the repro is a weird trait impl or the backtrace points to `rustc_trait_selection`, add `A-traits`).
- Add `T-*`, `WG-*`, `PG-*`, `F-*`, `requires-*`, and `regression-*` labels as appropriate.
    - Usually the ICE should be tagged `T-compiler`. If the issue concerns the type system (e.g. trait solver), also tag the issue with `T-types`.

> - If it is an ICE in rustdoc, add `T-rustdoc` instead of `T-compiler`.

Reviewer: Remark: there's *a bit* of nuance here, in that rustdoc is a rustc_private tool, which means that a root cause *can* be from the rustc frontend bits. But yes, I agree this is a better label in the general case, at worst the compiler label can be reapplied or adjusted.

## Further triaging

For issues that have been through the initial triaging step (that is, don't have the `needs-triage` label anymore), there are usually still things that can be improved. There are often many more labels that could be applied (using rustbot again if you don't have write access to [`rust-lang/rust`]).

Additionally, old (there is no clear definition of old yet, but something on the order of months) `S-needs-repro` issues can be closed if there is no way to make progress without a reproduction. This requires write access to [`rust-lang/rust`], but if you don't have them, you can just link the issue on Zulip (for example in [`t-release/triage`] or `general`) and someone with write access can close it for you.

> Another useful thing to do is to go through `E-needs-mcve` and `E-needs-bisection` issues and create minimizations or bisecting the issue (using [`cargo-bisect-rustc`]). When you provide one, you can also remove the label using rustbot (`@rustbot label -E-needs-bisection S-has-bisection`).

Reviewer: Nit:

```suggestion
Another useful thing to do is to go through `E-needs-mcve` and `E-needs-bisection` issues and create minimizations or bisecting the issue (using [`cargo-bisect-rustc`]). When you provide one, you should adjust the bisection status via rustbot (`@rustbot label -E-needs-bisection +S-has-bisection`).
```
